segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/ZooKeeperServiceRunner.java

```diff
@@ -63,7 +64,34 @@ public void start() throws Exception {
         serverFactory.configure(new InetSocketAddress(LOOPBACK_ADDRESS, this.zkPort), 1000);
         serverFactory.startup(s);
 
-        boolean b = LocalBookKeeper.waitForServerUp(address, LocalBookKeeper.CONNECTION_TIMEOUT);
+        boolean b = waitForServerUp(this.zkPort);
         log.info("ZooKeeper server {}.", b ? "up" : "not up");
     }
+
+    /**
+     * Blocks the current thread and awaits ZooKeeper to start running locally on the given port.
+     *
+     * @param zkPort The ZooKeeper Port.
+     * @return True if ZooKeeper started within a specified timeout, false otherwise.
+     */
+    public static boolean waitForServerUp(int zkPort) {
+        val address = LOOPBACK_ADDRESS.getHostAddress() + ":" + zkPort;
+        return LocalBookKeeper.waitForServerUp(address, LocalBookKeeper.CONNECTION_TIMEOUT);
+    }
+
+    public static void main(String[] args) throws Exception {
```

Comment:
Added a comment explaining that the purpose of this is to start ZK out of process.